File: vowels_substrings.c
// 2063. Vowels of All Substrings
#include <stdio.h>
#include <string.h>

/*
 * given a string 'word, return thesum of the number of vowels (a, e, i, o, u)
 * in every substring of 'word'. a substringis a contiguous (non-empty) sequence
 * of characters within a string. note, due to the large constraints, the answer
 * may not fit in a signed 32-bit integer. please be careful during
 * calculates...
 */

long long countVowels(char *word) {
  long long ans = 0;
  int n = strlen(word);
  for (int i = 0; i < n; i++)
    if (word[i] == 'a' || word[i] == 'e' || word[i] == 'i' || word[i] == 'o' ||
        word[i] == 'u')
      ans += (long long)(i + 1) * (n - i);
  return ans;
}

int main() {
  char w1[] = {"aba"}, w2[] = {"abc"}, w3[] = {"ltcd"};
  printf("%lld\n", countVowels(w1)); // expect: 6
  printf("%lld\n", countVowels(w2)); // expect: 3
  printf("%lld\n", countVowels(w3)); // expect: 0
}

File: vowels_substrings.py
# 2063. Vowels of All Substrings

"""
given a string 'word, return thesum of the number of vowels (a, e, i, o, u)
in every substring of 'word'. a substringis a contiguous (non-empty) sequence
of characters within a string. note, due to the large constraints, the answer
may not fit in a signed 32-bit integer. please be careful during
calculates...
"""


class Solution(object):
    def countVowels(self, word):
        ans, n = 0, len(word)
        d = {"a": 1, "e": 1, "i": 1, "o": 1, "u": 1}
        for i in range(n):
            if word[i] in d:
                ans += (n - i) * (i + 1)
        return ans


if __name__ == "__main__":
    obj = Solution()
    print(obj.countVowels("aba"))  # expect: 6
    print(obj.countVowels("abc"))  # expect: 3
    print(obj.countVowels("ltcd"))  # expect: 0
